### PR TITLE
Add check for building docsite

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
To make sure people don't break the build of the docsite

ping @oraNod 

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
 - Docs

### CODE REVIEW

**Review by tvo318 — APPROVED** [review]
I like it!

File: .github/workflows/docs.yml
Comment (oraNod) [inline .github/workflows/docs.yml:11]: How about using Fedora? One of my goals is to start using Fedora in all our tests. It'd be cool to start here.

File: .github/workflows/docs.yml
Comment (oraNod) [inline .github/workflows/docs.yml]: Do we really need this step? We can enable PR previews on Read The Docs when we set up hosting. Admittedly I've put this same step in other places but I'm thinking about disabling it.

If we do want to keep it, how about reducing the retention period so we don't have loads of doc builds hanging around and taking up space. The default is 90 days but we could drop that to ~7. https://github.com/actions/upload-artifact#retention-period

File: .github/workflows/docs.yml
Comment (AlanCoding) [reply .github/workflows/docs.yml:11]: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners

The choices are Ubuntu, Windows, or Mac.

File: .github/workflows/docs.yml
Comment (AlanCoding) [reply .github/workflows/docs.yml]: Oh I really wanted your feedback on this. Yes, I'm totally fine to get rid of this and even if we did keep it I agree with dropping the retention time.

I want to defer to whatever method will be used to publish them. So if the readthedocs integration is set up on their side, then it makes no sense to have this. I had in my head that maybe we would have this step, and elsewhere another step that uploaded it to the readthedocs API. That sounds unlikely so I'll push a commit to remove this.

File: .github/workflows/docs.yml
Comment (oraNod) [reply .github/workflows/docs.yml:11]: nuts. I could swear I noticed Fedora in another repo somewhere and it's kind of been on my to-do list to start looking at the switch. sorry for the noise.

File: .github/workflows/docs.yml
Comment (oraNod) [reply .github/workflows/docs.yml]: @AlanCoding Awesome. I can send a PR with the RTD config today and set up the server-side integration with you and TVo as maintainers. Then we can add it to the ansible namespace. Sound OK?

Comment (AlanCoding) [reply .github/workflows/docs.yml]: I don't understand all the proper nouns you reference there, but in general it sounds good. I assume that publishing these is next in your agenda and we can sync up in some other communication channels.

Comment (oraNod) [reply .github/workflows/docs.yml]: apologies, I have a habit of using "RTD" instead of "Read The Docs". I can ping you in Matrix to sync up on the publishing tomorrow if that works.

**Review by oraNod — APPROVED** [review]

**Pull request merged**
